proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java

```diff
@@ -348,29 +358,34 @@ public ProctorResult determineTestGroups(
             localContext.put(
                     UNITLESS_ALLOCATION_IDENTIFIER,
                     containsUnitlessAllocations ? VALUE_EXPRESSION_TRUE : VALUE_EXPRESSION_FALSE);
-            if (testChooser instanceof StandardTestChooser) {
-                final TestType testType = testChooser.getTestDefinition().getTestType();
-                if (testTypesWithInvalidIdentifier.contains(testType)
-                        && !containsUnitlessAllocations) {
+            final TestType testType = testChooser.getTestDefinition().getTestType();
+            if (testChooser instanceof UnitlessTestChooser) {
+                identifier = identifiers.getIdentifier(testType);
```

Comment:
will we always get empty string in this case?

Reply:
Good call out should probably force to empty string if it is null

Reply:
Thinking about this more should null be treated differently? I think no but we will need to make it a empty string to be able to hash it.